GeoLib/OctTree-impl.h

```diff
@@ -40,7 +40,16 @@ OctTree<POINT, MAX_POINTS> OctTree<POINT, MAX_POINTS>::createOctTree(T ll, T ur,
 			ur[1] += (dz-dy)/2.0;
 		}
 	}
-	return OctTree<POINT, MAX_POINTS>(ll, ur, eps);
+	if (eps == 0.0)
+		eps = std::numeric_limits<double>::epsilon();
+	for (std::size_t k(0); k<3; ++k) {
+		if (ur[k] - ll[k] > 0.0) {
+			ur[k] += (ur[k] - ll[k]) * 1e-6;
+		} else {
```

Comment:
do you mean there is a negative case `ur[k] - ll[k] < 0.0`? or only `ur[k] - ll[k] == 0.0`